### ISSUE TITLE
no-unmodified-loop-condition false positive with binary logical operators

### ISSUE THREAD

**Issue by l1bbcsg** [eslint/eslint#10327]
**Tell us about your environment**

* **ESLint Version:** `4.1.1`, `4.19.1`, `5.0.0-alpha.2`
* **Node Version:** `9.6.1`
* **npm Version:** `5.6.0`

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**
Just having `no-unmodified-loop-condition` on is enough.

<details>
<summary>Configuration (demo link)</summary>

https://eslint.org/demo/#eyJ0ZXh0IjoibGV0IGZvbyA9ICdmb28nO1xubGV0IGJhciA9ICdiYXInO1xuXG53aGlsZSAoZm9vID09PSAnZm9vJyAmJiBiYXIgPT09ICdiYXInKSB7XG5cdGJhciA9ICdiYXonO1xufVxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLXVubW9kaWZpZWQtbG9vcC1jb25kaXRpb24iOjJ9LCJlbnYiOnt9fX0=

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let foo = 'foo';
let bar = 'bar';

while (foo === 'foo' && bar === 'bar') {
	bar = 'baz';
}
```

**What did you expect to happen?**

No errors related to `no-unmodified-loop-condition`.

**What actually happened? Please include the actual, raw output from ESLint.**

>   4:8  error  'foo' is not modified in this loop  no-unmodified-loop-condition

**Comments**

While it's true `foo` is not modified, there's clearly nothing wrong (from syntax point of view) with that code and it does not cause infinite loop.

The rule description states:

>If a reference is inside of a binary expression or a ternary expression, this rule checks the result of the expression instead.

The result of binary expression `&&` did change inside the loop and the error is not valid. The same happens with `||` (`foo` should cast to `false` in that case or there will be infinite loop).

I think if ESLint cannot verify whether variable operand remaining unchanged results in change of expression result or not it should err on the side of caution, same as it does with property access and function call with this rule and avoid reporting this as a problem.
